test/index.test.ts

```diff
@@ -908,6 +908,33 @@ describe('Unit | Utility | changeset', () => {
         ]);
       });
 
+      it('can add an item to an index in an array where that item was previously removed', () => {
```

Comment:
this actually passes without the change to merge-deep -- I couldn't figure out how to reproduce the issue I was seeing from this kind of test -- I left this test in, because I think it's still valuable